src/effects/native/phasereffect.cpp

```diff
@@ -0,0 +1,199 @@
+    fb->setUnitsHint(EffectManifestParameter::UNITS_UNKNOWN);
+    fb->setDefault(0.0);
+    fb->setMinimum(-0.95);
+    fb->setMaximum(0.95);
+
+    EffectManifestParameter* range = manifest.addParameter();
+    range->setId("range");
+    range->setName(QObject::tr("Range"));
+    range->setDescription("Controls the frequency range across which the notches sweep.");
+    range->setControlHint(EffectManifestParameter::CONTROL_KNOB_LINEAR);
+    range->setSemanticHint(EffectManifestParameter::SEMANTIC_UNKNOWN);
+    range->setUnitsHint(EffectManifestParameter::UNITS_UNKNOWN);
+    range->setDefault(0.5);
+    range->setMinimum(0.05);
+    range->setMaximum(0.95);
+
+    return manifest;
+}
+
+PhaserEffect::PhaserEffect(EngineEffect* pEffect,
+                           const EffectManifest& manifest) 
+        : m_pStagesParameter(pEffect->getParameterById("stages")),
+          m_pLFOFrequencyParameter(pEffect->getParameterById("lfo_frequency")),
+          m_pDepthParameter(pEffect->getParameterById("depth")),
+          m_pFeedbackParameter(pEffect->getParameterById("feedback")),
+          m_pRangeParameter(pEffect->getParameterById("range")),
+          m_pStereoParameter(pEffect->getParameterById("stereo")) {
+    Q_UNUSED(manifest);
+}
+
+PhaserEffect::~PhaserEffect() {
+    //qDebug() << debugString() << "destroyed";
+}
+
+void PhaserEffect::processChannel(const ChannelHandle& handle,
+                                  PhaserGroupState* pState,
+                                  const CSAMPLE* pInput, CSAMPLE* pOutput,
+                                  const unsigned int numSamples,
+                                  const unsigned int sampleRate,
+                                  const EffectProcessor::EnableState enableState,
+                                  const GroupFeatureState& groupFeatures) {
+
+    Q_UNUSED(handle);
+    Q_UNUSED(enableState);
+    Q_UNUSED(groupFeatures);
+    Q_UNUSED(sampleRate);
+
+    CSAMPLE frequency = m_pLFOFrequencyParameter->value();
+    CSAMPLE depth = m_pDepthParameter->value();
+    CSAMPLE feedback = m_pFeedbackParameter->value();
+    CSAMPLE range = m_pRangeParameter->value();
+    int stages = m_pStagesParameter->value();
+
+    CSAMPLE* oldInLeft = pState->oldInLeft;
+    CSAMPLE* oldOutLeft = pState->oldOutLeft;
+    CSAMPLE* oldInRight = pState->oldInRight;
+    CSAMPLE* oldOutRight = pState->oldOutRight;
+
+    //Using two sets of coefficients for left and right channel 
+    CSAMPLE filterCoefLeft = 0;
+    CSAMPLE filterCoefRight = 0;
+
+    //If changing the 'Rate' knob, the phaser will use the initial parameters    
+    //until a desired frequency is chosen (to avoid bubbling noises)
+    if (frequency != pState->frequency) {
+        pState->wait = 0;
+        pState->frequency = frequency; 
+        pState->change = true;
+    }
+
+    if (pState->change) {
+        pState->wait++;
+        if (pState->wait == pState->limit) {
```

Comment:
I do not understand how it works. Playing with the rate knob it looks like it works sometimes. 
If you move the knob continuously the new value seams to be not adopted. 
It looks like the root issue is below. See below ...